### PR TITLE
ON-39390 # don't fetch dyamic options in read only forms if element c…

### PR DESCRIPTION
**Pull request by divporter**
…ontains options

### CODE REVIEW

File: src/hooks/useDynamicOptionsLoaderState.tsx [hunk 1]
```diff
@@ -166,25 +169,37 @@ const LoadOptionsSet = React.memo(function LoadOptionsSet({
...
   const hasOptionsSet = React.useMemo<boolean>(() => {
     return !!formElementsService.findFormElement(
       form.elements,
       (formElement) => {
         const formElementWithOptions =
           typeCastService.formElements.toOptionsElement(formElement)
+
+        // for read only forms we don't want to fetch if we don't have to
+        if (
+          formIsReadOnly &&
```
Comment (mymattcarroll) [inline src/hooks/useDynamicOptionsLoaderState.tsx:188]: I dont actually think we need this... 🤔

Comment (divporter) [reply src/hooks/useDynamicOptionsLoaderState.tsx:188]: mmm you might be right. Will a clarification use the previous definition or the current form def?

Comment (mymattcarroll) [reply src/hooks/useDynamicOptionsLoaderState.tsx:188]: clarification will use current form def

Comment (divporter) [reply src/hooks/useDynamicOptionsLoaderState.tsx:188]: removing...

File: src/hooks/useDynamicOptionsLoaderState.tsx [hunk 2]
```diff
@@ -177,8 +177,18 @@ const LoadOptionsSet = React.memo(function LoadOptionsSet({
...
           typeCastService.formElements.toOptionsElement(formElement)
+
+        // for read only forms we don't want to fetch if we don't have to
+        if (
+          formElementWithOptions?.optionsType === 'DYNAMIC' &&
+          formElementWithOptions.options?.length
+        ) {
+          return false
+        }
+
```
Comment (mymattcarroll) [inline src/hooks/useDynamicOptionsLoaderState.tsx:188]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- No longer fetch dyamic options in read only forms if element contains options
+
+### Changed
```
Comment (mymattcarroll) [inline CHANGELOG.md:14]: can remove extra changed

**Review by mymattcarroll — APPROVED** [review]

**Pull request merged**